python/mlc_chat/serve/entrypoints/openai_entrypoints.py

```diff
@@ -341,12 +431,43 @@ async def completion_stream_generator() -> AsyncGenerator[str, None]:
     assert finish_reason is not None
 
     async_engine.record_event(request_id, event="finish")
+
+    if conv_template.use_function_calling:
+        try:
+            fn_json_list = convert_function_str_to_json(output_text)
+        except (SyntaxError, ValueError):
+            return entrypoint_utils.create_error_response(
+                HTTPStatus.BAD_GATEWAY, message="Got an invalid function call output from model"
+            )
```

Comment:
we should look into the OpenAI spec on what reply to send

Reply:
Ideally we should not fail when the error was due to model failures. In future we can retry. We can also add a "finish_reason" =  "error" that streams back error message. 

Likely that would enable easier handling of the errors

Reply:
Ideally, we would want a model that can atleast output well formed json output even if it is not correct. For openAI, they don't have an exact specification for errors in case of wrong model output. Generally, if the query is such that it can be answered as a function call, it switches to text output probably using a different model (using maybe some moe at the backend).

Reply:
We can change the code to move the model output to text as well. But it might not make sense like the openAI output and would rather just be a malformed function call

Reply:
Let us do a reply with `finish_reason` field as "error", and the output text being the error message